assembly/src/parsers/ast/mod.rs

```diff
@@ -99,7 +99,7 @@ impl ModuleAst {
     }
 
     /// Returns a `ModuleAst` struct by its byte representation.
-    pub fn from_bytes(bytes: &mut &[u8]) -> Result<Self, SerializationError> {
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, SerializationError> {
```

Comment:
Good catch! We can also probably optimize `ByteReader` to have it work with slices rather than with vectors. Maybe create an issue for this?